### ISSUE TITLE
add additional species names to gene labels

### ISSUE THREAD

**Issue by balhoff** [geneontology/neo#116]
See https://github.com/geneontology/go-site/issues/1955#issuecomment-1529033344

Final list is on the tab named ‘List V2’:
 
https://docs.google.com/spreadsheets/d/1dQ0gN2HQjq2IFuXQrUdE5uRmMvZaUuylYQ1RvclrCqc/edit?gid=1998866383#gid=1998866383

**Comment by pgaudet** [comment]
Species names are still not being displayed in Noctua. 

For example, Taxon:42789 & model: http://noctua.geneontology.org/workbench/noctua-visual-pathway-editor/?model_id=gomodel%3A6494e2e900000134

**Comment by balhoff** [comment]
This seems to be controlled by the script https://github.com/geneontology/neo/blob/master/gpi2obo.pl. When it is called by the Makefile, a species code needs to be passed to be appended to the gene product name. There seems to be some unfinished work related to handling virus names here:

https://github.com/geneontology/neo/blob/b1a10399234c2638250286f8434093e494ccc100/Makefile#L41-L68

**Comment by kltm** [comment]
(Quiet shoutout to https://github.com/geneontology/project-management/issues/52)

**Comment by pgaudet** [comment]
Trying to quote @kltm 

How the date gets in: 
- takes the GPI file, and convert it into an obo and load it

**Comment by kltm** [comment]
Noting that this seems to be the "next step" referred to in https://github.com/geneontology/neo/issues/77. Essentially, we stopped there before getting to this point.

**Comment by pgaudet** [comment]
Right now there is a single file per file - so this is complicated to load

**Comment by kltm** [comment]
There are a few ways to deal with this. It seems that the codes mostly come from, by way of a JSON derivative, the metadata/datasets YAML files. I'm not sure there's much to do there for adding a bunch of additional species.

It would be nice if we could  just modify `build-neo-makefile.py` with an optional override file, but that too seems to operate on a file-per-species basis. 

The most direct way, without redoing a bunch of what we're doing, might be to add GPI files for the species that we want and adding the metadata for them in datasets.

**Comment by pgaudet** [comment]
GPIs for the [organisms requested by Patrick Masson and Paul Denny ](https://docs.google.com/spreadsheets/d/1dQ0gN2HQjq2IFuXQrUdE5uRmMvZaUuylYQ1RvclrCqc/edit?gid=0#gid=0)are now being generated by GOA at each release: 
https://ftp.ebi.ac.uk/pub/contrib/goa/virus_bacteria_gpi4neo/

**Comment by vanaukenk** [comment]
Thanks @kltm @pgaudet @alexsign

@kltm - on today's workbenches call I'd just like to confirm how/when/who should test this.

**Comment by vanaukenk** [comment]
From 2024-01-30 workbenches call:

Once the new species are added to neo, we'll test on the next Noctua maintenance outage.

**Comment by kltm** [comment]
Data needed for this now being populated to GO mirror of GOA data for our pipeline.

**Comment by kltm** [comment]
Okay, I did a little testing of this NEO data load on amigo-staging (soon reverting), and it looks like there is a little more work to be done

- [ ] it looks like the data did not automatically propagate as I expected; I will need to explore the NEO build a little more to see what went wrong
- [ ] something that we will have to fix either way is adding the proper species names to the metadata files; specifically, it would be adding the `species_code` field (e.g `species_code: Ggal`) to each of the entries; @pgaudet should I just use the standard shortening here, adding from NCBITaxon, or does something want something else there for nomenclature?

**Comment by pgaudet** [comment]
Hi @kltm 

Are you limited to 4 characters? Ideally we would align to the UniProt 5 characters species_code.

**Comment by Pauldenny** [comment]
If it's possible @kltm @pgaudet  the UniProt 5-character code would be preferable.

**Comment by kltm** [comment]
@Pauldenny For clarification, would this be the canonical source for the 5-character code?
https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/docs/speclist.txt

**Comment by cmungall** [comment]
5 letter uniprot code is good!

**Comment by Pauldenny** [comment]
I believe that's the source you should use, @kltm

**Comment by kltm** [comment]
@Pauldenny The above file mostly works, but is missing two entries for our purposes:

https://www.ncbi.nlm.nih.gov/datasets/taxonomy/36352/
https://www.ncbi.nlm.nih.gov/datasets/taxonomy/128958/

I've used placeholders for the time being.

**Comment by kltm** [comment]
Talking to @tmushayahama , we will need to have a release of the Pathway Viewer that is more clever at removing the "species" part of the label. This will need to be done before we release the new data.

**Comment by kltm** [comment]
To properly lay out the issues and options here, it turns out that the Pathway Viewer widget uses a statically compiled file to filter out the species part of the label. These are introduced by the NEO data load taken by minerva and propagates to the API and other locations by way of the model JSON.

- If we proceed without a widget release (which would allow us to get the new species in quickly), things would continue to work as they do now, but models with new species' gps would have the 5-letter code exposed. In this case, while this is annoying, but we wouldn't consider it a blocking issue.
- If we wait until the code is fixed to be a little more flexible and dynamic, we would also want to make sure all of our users have the new widget before we make this data update. This could be a slower process, and we would likely not be aiming to get this done at the next outage. In this case, https://github.com/geneontology/wc-gocam-viz/issues/73 would be a blocking issue.

**Comment by Pauldenny** [comment]
Hi @kltm thanks for explainer - I would prefer to get the new species in quickly, if possible and workaround the gene naming

**Comment by kltm** [comment]
There is a collision between uniprot_reviewed.gpi.gz and taxon_12118.gpi.gz; it looks like they have difference names for the same identifier (from @balhoff ), likely around "name( P03305 FMDVO)" and "name( P03305 NCBITaxon:73482)". For expediency of testing, I'm going to remove taxon_12118.gpi.gz from the build for the moment to see if we can get more progress.

**Comment by kltm** [comment]
Okay, there seem to be multiple issues. Unfortunately, it stops when hitting the first rather than continuing, so we'll have to take a few passes at this. I will keep a list of issues as I find them here:

- UniProtKB:P03305 in taxon_12118.gpi.gz
- UniProtKB:P0DPR4 in taxon_470.gpi.gz

I will be eliminating the files as I go; then examine the problematic files individually.

**Comment by kltm** [comment]
@pgaudet As examples from for full lines, we have the following two sets, which are likely the problematic lines. While it doesn't seem to be the name, I can see that the divergences would be an issue.

taxon_470.gpi.gz
```
UniProtKB	P0DPR4	abaQ	Quinolone resistance transporter	abaQ|AUO97_17040	protein	taxon:470			db_subset=Swiss-Prot
```
uniprot_reviewed.gpi.gz
```
UniProtKB	P0DPR4	abaQ	Quinolone resistance transporter	abaQ|AUO97_17040	protein	taxon:400667		EMBL:CP018664	db_subset=Swiss-Prot|taxon_name=Acinetobacter baumannii (strain ATCC 17978 / CIP 53.77 / LMG 1025 / NCDC KC755 / 5377)|proteome=complete
```

taxon_11218.gpi.gz
```
UniProtKB	P03305	P03305	Genome polyprotein		protein	taxon:12118			db_subset=Swiss-Prot
```
uniprot_reviewed.gpi.gz
```
UniProtKB	P03305	P03305	Genome polyprotein		protein	taxon:73482		EMBL:J02185	db_subset=Swiss-Prot|taxon_name=Foot-and-mouth disease virus (isolate Bovine/Germany/O1Kaufbeuren/1966 serotype O)|taxon_common_name=FMDV|proteome=complete
```

Solutions would either be removing the offending entries from one or both files, or regenerating the files so that the information is aligned.

**Comment by cmungall** [comment]
Is there any reason to load uniprot_reviewed (swissprot) now? If we need a species we can just add it. It is not as if we got great coverage from swissprot, e.g. many species have a single gene, not much use for GO-CAMs

**Comment by kltm** [comment]
@pgaudet If I could get clarification on https://github.com/geneontology/neo/issues/116#issuecomment-2652423042, that might help the next trial I do for this load.
That said, I'm going to try and go back to our "traditional" load for tomorrow, but will continue working on the new load. Once we have that working and the correct included files, we can have a special outage to include them (or add them in the next outage this month).

**Comment by kltm** [comment]
@pgaudet @vanaukenk We now have a NEO test load with the taxon files, but with the uniprot_reviewed removed (see https://github.com/geneontology/neo/issues/116#issuecomment-2652423042) at:

https://amigo-staging.geneontology.io/amigo/search/ontology (new load)

versus the current 

http://noctua-amigo.berkeleybop.org/amigo/search/ontology (current load)

About ~3,000,000 vs ~2,000,000 terms.

**Comment by pgaudet** [comment]
> Is there any reason to load uniprot_reviewed (swissprot) now? If we need a species we can just add it. It is not as if we got great coverage from swissprot, e.g. many species have a single gene, not much use for GO-CAMs

Can anyone get the different taxa currently used in GO-CAM? I think we may interacting viruses and bacteria. As I understood it, the load of the 62 new species was only to be able to get the gene names correctly on the activity units. But I dont think 'just adding' new species is so easy. Can we not keep the other uniprot_reviewed (swissprot) ?

**Comment by pgaudet** [comment]
Confirmed with @kltm : new species loaded will use the UniProt 5-letter code. 

If we want to switch all species, this will need some changes in the code.

**Comment by kltm** [comment]
@pgaudet These files do not exist in the given directory:
```
[2025-03-06T08:59:15.567Z] No such file ‘taxon_574.gpi.gz’.
[2025-03-06T08:59:27.839Z] No such file ‘taxon_10299.gpi.gz’.
[2025-03-06T08:59:29.592Z] No such file ‘taxon_10338.gpi.gz’.
[2025-03-06T08:59:31.144Z] No such file ‘taxon_10360.gpi.gz’.
[2025-03-06T08:59:44.013Z] No such file ‘taxon_93061.gpi.gz’.
[2025-03-06T09:00:11.907Z] No such file ‘taxon_295027.gpi.gz’.
[2025-03-06T09:00:13.660Z] No such file ‘taxon_301447.gpi.gz’.
[2025-03-06T09:00:22.974Z] No such file ‘taxon_565655.gpi.gz’.
[2025-03-06T09:00:26.479Z] No such file ‘taxon_587200.gpi.gz’.
```
I'm, removing them for the moment from the processing. Let me know when these have been added and I will add them back in.

**Comment by kltm** [comment]
@pgaudet @vanaukenk We now have a NEO test load with a large subset (see https://github.com/geneontology/neo/issues/116#issuecomment-2704658900) of the taxon files at:

https://amigo-staging.geneontology.io/amigo/search/ontology (new load)

versus the current

http://noctua-amigo.berkeleybop.org/amigo/search/ontology (current load)

2069053 versus 1885040 terms.

**Comment by kltm** [comment]
@alexsign has provided updated files that align with what @pgaudet was looking for. Will update accordingly.

**Comment by pgaudet** [comment]
Is there anything we should/can QC ?

**Comment by kltm** [comment]
Rolling out onto amigo-staging right now.

**Comment by vanaukenk** [comment]
@kltm @pgaudet 

I did some testing before tomorrow's outage by comparing search results on amigo vs amigo-staging.  I can see a number of the new species now on amigo-staging with the five-letter species code visible in the search.

I also found at least one E. coli reviewed UniProt entry that is in amigo but not now in staging:  UniProtKB:P21513 
Does it make sense that we would have lost any reviewed E. coli K12 entries?  Am I just missing it on staging?

**Comment by kltm** [comment]
@vanaukenk Well, I'm not sure it matters as I've only added data--I have not removed any. If we go back to the previous load, that would still be missing. I think if anything is not there that should be, we should treat that as a separate issue.
That said, if the missing entries are disruptive, we can continue without an update.

**Comment by pgaudet** [comment]
@kltm where do we get Ecoli entities from ? 
That ID is in the EcoCyc GPI (https://www.ai.sri.com/~kr/go/latest/ecocyc.gpi.gz), listed in their yaml: 
https://github.com/geneontology/go-site/blob/master/metadata/datasets/ecocyc.yaml

Thanks, Pascale

**Comment by cmungall** [comment]
This is no different from mouse using MGI IDs, pombe using PomBase IDs, etc

**Comment by kltm** [comment]
@pgaudet Looks like:
```
wget --no-check-certificate https://www.ai.sri.com/~kr/go/latest/ecocyc.gpi.gz
```
I'd note that it's in the GPI as a synonym; it can be "found" in the NEO load on staging using that, but the direct identifier is in the EcoCyc namespace.

**Comment by Pauldenny** [comment]
OK - one point of clarification, @pgaudet - when you write: "note that it's in the GPI as a synonym" - is the "it" the UniProt accession?
 @cmungall - guess I'm just being parochial / UniProt-centric

**Comment by pgaudet** [comment]
> I'd note that it's in the GPI as a synonym; it can be "found" in the NEO load on staging using that, but the direct identifier is in the EcoCyc namespace.

Still that ID should be 'findable' just as we can find C elegans genes in Noctua when searching with a UniProt ID. 

Perhaps the work that needs to be done is to change the primary ID for the E coli genes already annotated? (Note that this only affect NCBITaxon:83333). 

Thanks, Pascale

**Comment by vanaukenk** [comment]
@kltm @pgaudet 

So far what I've checked comparing amigo, staging-amigo, noctua-amigo, and noctua makes sense, but I'm only spot-checking individual genes and taxa.

I'm still not sure how to assess the overall drop.  Is it possible to diff the suspected files to spot check that, as well?  I'm open to other suggestions for testing that part.

Thx.

**Comment by vanaukenk** [comment]
@pgaudet @kltm 
Trying to follow-up on this ticket, but I can't find where NCBITaxon:42789 is currently being loaded into NEO.

Let's check on this tomorrow.

**Comment by kltm** [comment]
I would support closing this and opening new tickets if there are additions or problems found.
